### PR TITLE
Fix mlflowdbfs availability check

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Fix mlflowdbfs availability check on Databricks; an equivalent patch has been submitted for the Databricks Machine Learning Runtime with MLflow 1.28.0.

## How is this patch tested?

Included integration tests

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/spark.py [hunk 1]
```diff
@@ -418,12 +418,26 @@ def _should_use_mlflowdbfs(root_uri):
     # the Hadoop FileSystem API until a read call has been issued.
     from mlflow.utils._spark_utils import _get_active_spark_session
 
+    if (
+        not is_valid_dbfs_uri(root_uri)
+        or not is_databricks_acled_artifacts_uri(root_uri)
+        or not databricks_utils.is_in_databricks_runtime()
+        or environment_variables._DISABLE_MLFLOWDBFS.get() not in ["", "False", "false"]
```
Comment (harupy) [inline mlflow/spark.py:425]:
```suggestion
        or environment_variables._DISABLE_MLFLOWDBFS.get().lower() not in ["", "false"]
```

Comment (harupy) [reply mlflow/spark.py:425]:
```suggestion
        or environment_variables._DISABLE_MLFLOWDBFS.get().lower() == "true"
```

Is this more straightforward?

File: mlflow/spark.py [hunk 2]
```diff
@@ -418,12 +418,26 @@ def _should_use_mlflowdbfs(root_uri):
...
     except Exception as e:
         # The load invocation is expected to throw an exception.
-        if "MlflowdbfsClient" in str(e):
-            # The subsequent logic used to determine mlflowdbfs availability on Databricks
+        mlflowdbfs_read_exception_str = str(e)
+
+    try:
+        return _HadoopFileSystem.is_filesystem_available(_MLFLOWDBFS_SCHEME)
+    except Exception:
+        if "MlflowdbfsClient" in (mlflowdbfs_read_exception_str or ""):
```
Comment (harupy) [inline mlflow/spark.py:439]:
```suggestion
        return "MlflowdbfsClient" in (mlflowdbfs_read_exception_str or "")
```

Can we just return `"MlflowdbfsClient" in (mlflowdbfs_read_exception_str or "")`?

File: mlflow/spark.py (hunk 1 above)
Comment (dbczumar) [reply mlflow/spark.py]: It definitely is! Committed your second suggestion

File: mlflow/spark.py (hunk 2 above)
Comment (dbczumar) [reply mlflow/spark.py]: Doh! Absolutely! Done!

**Comment by dbczumar** [comment]
Thanks @harupy , I've addressed your comments!

**Review by harupy — APPROVED** [review]
LGTM!

File: tests/spark/test_spark_model_export.py [hunk 3]
```diff
@@ -640,27 +640,67 @@ def test_model_is_recorded_when_using_direct_save(spark_model_iris):
...
     monkeypatch,
     spark_model_iris,
+    artifact_uri,
     db_runtime_version,
     mlflowdbfs_disabled,
     mlflowdbfs_available,
     expectedURI,
 ):
+    def mock_spark_session_load(path):
+        raise Exception("MlflowDbfsClient operation failed!")
```
Comment (harupy) [inline tests/spark/test_spark_model_export.py:694]: Interesting. `path` is not used but pylint doesn't seem to complain lol.

Comment (harupy) [reply tests/spark/test_spark_model_export.py:694]: Never mind this comment.

**Pull request merged**